src/Tools/dotnet-monitor/InProcessFeatures/InProcessFeatureOptionsBinder.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.Options;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal static class InProcessFeatureOptionsBinder
+    {
+        public static void BindEnabled(IInProcessFeatureOptions options, string configurationKey, IConfiguration configuration, bool enabledByDefault)
+        {
+            bool? inProcessFeaturesEnabled = null;
+            IConfigurationSection inProcessFeaturesSection = configuration.GetSection(ConfigurationKeys.InProcessFeatures_Enabled);
+            if (!string.IsNullOrEmpty(inProcessFeaturesSection.Value))
+            {
+                inProcessFeaturesEnabled = inProcessFeaturesSection.Get<bool>();
+            }
+
+            // If in-process features are disabled, then disable the individual feature
+            if (!inProcessFeaturesEnabled.GetValueOrDefault(true))
```

Comment:
I'm struggling a bit with the readability flow in this method. Is it possible we could do some nesting up here? My understanding is that `inProcessFeaturesEnabled` will only have a value if the above conditional is met, so we could nest this into there. Additionally, my understanding is that `options.Enabled.HasValue` will only be satisfied if the current condition is met since that's the only place `options.Enabled` is being explicitly set thus far, so we could combine that logic as well. Feel free to correct me if I mis-read this, but I just noticed as I was reviewing this that it was requiring a lot more cognitive effort than I'd anticipated to keep everything straight.

Reply:
Both of the Enable properties (at the InProcessFeatures level and the individual feature level) are nullable. When one of them is null, it is deferring the final decision of whether a feature is enable or not to the other property. So here, either the user didn't specify anything (the value is null and thus deferring whether individual features are enabled to their individual Enabled properties), the user specified `true` (so we don't disable all individual features), or the user specified `false` (so we disable all individual features).

The `!inProcessFeaturesEnabled.GetValueOrDefault(true)` is effectively asking "should we disable all features?":

| `inProcessFeaturesEnabled` | `!inProcessFeaturesEnabled.GetValueOrDefault(true)` | Explanation |
|---|---|---|
| null | false | User didn't specify anything -> don't disable all features |
| true | false | User said to enable in-process features -> don't disable all features |
| false | true | User said to disable in-process features -> disable all features |

Reply:
I should write some tests for the InProcessFeatureOptionsBuilder to validate each path within this function; that'll make sure it doesn't change accidentally and should make it easier to understand the intended outcome for each input combination.